### ISSUE TITLE
Add support for multiple media types

### ISSUE THREAD

**Issue by vmolsa** [michael/github#218]
repo.read() function should have option for choosing the format of the receiving data. (base64)
For example, if requested target is image file and if you are going to show it on browser then it could be much easier to download content as base64 image so the image url could be easily generated when there is no need to convert it to base64 anymore.. 

and for receiving / sending commits, pulls, etc.. it would be nice to have an option choosing for diff / patch media type. 

https://developer.github.com/v3/media/

https://developer.github.com/changes/2012-12-10-Diff-and-patch-media-types/

**Comment by aendrew** [comment]
Marking as a mid-priority item.

**Comment by aendrew** [comment]
#219 should fix this, just waiting on tests to merge.
